apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -21,6 +32,29 @@ export class GetWorkflowUseCase {
       })
     );
 
-    return toResponseWorkflowDto(workflowEntity);
+    const fullSteps = await this.getFullWorkflowSteps(workflowEntity, command.user);
+
+    return toResponseWorkflowDto(workflowEntity, fullSteps);
+  }
+
+  private async getFullWorkflowSteps(
+    workflow: WorkflowInternalResponseDto,
+    user: UserSessionData
+  ): Promise<StepDataDto[]> {
+    const stepPromises = workflow.steps.map((step: NotificationStepEntity & { _id: string }) =>
+      this.buildStepDataUsecase.execute(
+        BuildStepDataCommand.create({
+          workflowIdOrInternalId: workflow._id,
+          stepIdOrInternalId: step._id,
+          user,
+        })
+      )
+    );
```

Comment:
maybe we could throw it here? this way we will have the step id that failed as well.

```suggestion
try{
      this.buildStepDataUsecase.execute(
        BuildStepDataCommand.create({
          workflowIdOrInternalId: workflow._id,
          stepIdOrInternalId: step._id,
          user,
        })
      )
    );
    } catch (error) {
      throw new InternalServerErrorException({ message: 'Failed to build full workflow steps:', error: error.message, workflowId: workflow._id, stepId: step._id );
    }
```